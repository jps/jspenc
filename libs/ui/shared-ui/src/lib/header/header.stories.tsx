import { Meta } from '@storybook/react';
import { BaseProvider } from '../..';
import { Header } from './header';

export default {
  component: Header,
  title: 'Header',
} as Meta;

export const HeaderDefault = () =>
  <BaseProvider>
    <Header />
  </BaseProvider>;