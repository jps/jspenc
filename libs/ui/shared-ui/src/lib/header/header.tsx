import {
  Block,
  IconFilledGitHub,
  Stack,
  IconButton,
  IconFilledTwitter,
  styled,
  getSpacingCssFromTheme,
  toNewsKitIcon,
} from 'newskit';
import { WeatherMoon } from '@emotion-icons/fluentui-system-filled/WeatherMoon';
import { WeatherSunny } from '@emotion-icons/fluentui-system-filled/WeatherSunny'
import Link from 'next/link';
import { ReactNode, useContext } from 'react';
import { SiteThemeContext } from '../..';
import { Logo } from '../logo/logo';
import { JspencThemeLight } from '../theme/theme';

const HeaderContainer = styled.header`
  ${getSpacingCssFromTheme('marginTop', 'space040')};
  ${getSpacingCssFromTheme('marginBottom', 'space040')};
`;

const defaultButtonStyle = {
  overrides: { stylePreset: 'iconButtonMinimalSecondary' },
};
const defaultIconSize = { overrides: { size: 'iconSize030' } };

const MoonIcon = toNewsKitIcon(WeatherMoon);
const SunIcon = toNewsKitIcon(WeatherSunny);

const DayNightModeToggle = () => {
  const { currentTheme, toggleTheme } = useContext(SiteThemeContext);
  const isLightTheme = currentTheme.name === JspencThemeLight.name;
  
  const createButton = (
    children: ReactNode,
    ariaLabel: string  ) => (
    <IconButton
      children={children}
      aria-label={ariaLabel}
      onClick={() => {
        toggleTheme();
      }}
      {...defaultButtonStyle}
    />
  );

  return isLightTheme
    ? createButton(
        <MoonIcon {...defaultIconSize} />,
        'enable night mode'
      )
    : createButton(
        <SunIcon {...defaultIconSize} />,
        'enable day mode'
      );
};

export const Header = () => (
  <HeaderContainer>
    <Stack flow="horizontal-center" stackDistribution="space-between">
      <Block>
        <Link href="/" passHref>
          {/* This href attribute isn't technically needed but gets around linter warnings because of next.js's weird api */}
          <a href="/">
            <Logo title="Go to James Spencer home" />
          </a>
        </Link>
      </Block>
      <Block>
        <Stack flow="horizontal-center" spaceInline="space030">
          <IconButton
            aria-label="github"
            {...defaultButtonStyle}
            href="https://github.com/jps"
          >
            <IconFilledGitHub {...defaultIconSize} />
          </IconButton>
          <IconButton
            aria-label="twitter"
            {...defaultButtonStyle}
            href="https://twitter.com/jpspenc"
          >
            <IconFilledTwitter {...defaultIconSize} />
          </IconButton>
          <DayNightModeToggle />
        </Stack>
      </Block>
    </Stack>
  </HeaderContainer>
);
