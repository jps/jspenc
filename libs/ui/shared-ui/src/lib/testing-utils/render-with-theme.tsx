import { render } from '@testing-library/react';
import { BaseProvider } from '../..';

export const renderWithTheme =
    (ui:React.ReactElement) => render(<BaseProvider>{ui}</BaseProvider>);