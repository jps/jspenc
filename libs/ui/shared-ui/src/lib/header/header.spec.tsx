import { renderWithTheme } from '../teting-utils/render-with-theme';

import { Header } from './header';

describe('Header', () => {
  it('should render successfully', () => {
    const { baseElement } = renderWithTheme(<Header />);
    expect(baseElement).toMatchSnapshot();
  });
});
