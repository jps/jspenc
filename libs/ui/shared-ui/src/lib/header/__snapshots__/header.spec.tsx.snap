// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render successfully 1`] = `
<body>
  <div>
    <header
      class="css-1qgia4e"
    >
      <div
        class="css-1bq9n3y"
      >
        <div
          class="css-30qa7x"
        >
          <a
            href="/"
          >
            <svg
              fill="none"
              height="80"
              role="img"
              viewBox="0 0 120 120"
              width="80"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                Go to James Spencer home
              </title>
              <rect
                fill="#9DB7D3"
                fill-opacity="0.5"
                height="90"
                width="110"
                x="10"
                y="20"
              />
              <rect
                fill="#1C7293"
                fill-opacity="0.5"
                height="110"
                width="90"
                x="20"
                y="10"
              />
              <path
                d="M0 0H100V100H0V0Z"
                fill="#D5DCF9"
                fill-opacity="0.5"
              />
              <path
                d="M23.736 88.56C24.36 88.992 25.464 89.52 27.048 90.144C28.632 90.72 30.504 91.008 32.664 91.008C35.208 91.008 37.128 90.456 38.424 89.352C39.72 88.2 40.584 86.424 41.016 84.024C41.448 81.624 41.664 78.504 41.664 74.664V47.88H51.6V73.872C51.6 77.712 51.384 81.216 50.952 84.384C50.52 87.552 49.68 90.288 48.432 92.592C47.184 94.896 45.288 96.672 42.744 97.92C40.248 99.168 36.912 99.792 32.736 99.792C30.432 99.792 28.368 99.528 26.544 99C24.72 98.472 23.04 97.68 21.504 96.624L23.736 88.56ZM92.1512 61.272C91.7192 60.792 90.8072 60.168 89.4152 59.4C88.0232 58.584 86.3432 57.864 84.3752 57.24C82.4552 56.568 80.4632 56.232 78.3992 56.232C72.9752 56.232 70.2632 58.152 70.2632 61.992C70.2632 63.384 70.6952 64.488 71.5592 65.304C72.4232 66.12 73.7192 66.816 75.4472 67.392C77.1752 67.968 79.3352 68.592 81.9272 69.264C85.2872 70.128 88.1912 71.136 90.6392 72.288C93.1352 73.44 95.0312 74.976 96.3272 76.896C97.6712 78.768 98.3432 81.264 98.3432 84.384C98.3432 88.032 97.4552 90.984 95.6792 93.24C93.9512 95.448 91.6472 97.056 88.7672 98.064C85.8872 99.072 82.7672 99.576 79.4072 99.576C75.5672 99.576 71.7512 99 67.9592 97.848C64.2152 96.648 60.8792 95.016 57.9512 92.952L62.3432 84.384C62.7752 84.816 63.5432 85.392 64.6472 86.112C65.7512 86.784 67.0952 87.48 68.6792 88.2C70.2632 88.872 71.9912 89.448 73.8632 89.928C75.7352 90.408 77.6312 90.648 79.5512 90.648C85.0232 90.648 87.7592 88.896 87.7592 85.392C87.7592 83.904 87.2072 82.704 86.1032 81.792C84.9992 80.88 83.4392 80.112 81.4232 79.488C79.4552 78.864 77.1512 78.168 74.5112 77.4C71.2472 76.488 68.5352 75.48 66.3752 74.376C64.2632 73.224 62.6792 71.808 61.6232 70.128C60.5672 68.4 60.0392 66.216 60.0392 63.576C60.0392 60.072 60.8552 57.144 62.4872 54.792C64.1672 52.392 66.4232 50.592 69.2552 49.392C72.0872 48.144 75.2312 47.52 78.6872 47.52C82.2392 47.52 85.5512 48.096 88.6232 49.248C91.6952 50.4 94.3352 51.72 96.5432 53.208L92.1512 61.272Z"
                fill="#161032"
              />
            </svg>
          </a>
        </div>
        <div
          class="css-30qa7x"
        >
          <div
            class="css-u7u3qq"
          >
            <div
              class="css-fcx7ij"
            >
              <a
                aria-label="github"
                class="css-5p18y4"
                data-testid="buttonLink"
                href="https://github.com/jps"
              >
                <svg
                  aria-hidden="true"
                  class="css-1w2y819"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.9990792,2 C6.47773105,2 2,6.47565999 2,11.9970521 C2,16.4137976 4.86503576,20.1603908 8.83876117,21.4822847 C9.33908346,21.5743385 9.5214095,21.2656512 9.5214095,21.000536 C9.5214095,20.7636506 9.512815,20.1346157 9.50790386,19.3006075 C6.72635747,19.904481 6.13947635,17.9603029 6.13947635,17.9603029 C5.68458209,16.8053335 5.02894503,16.4978735 5.02894503,16.4978735 C4.12099819,15.878044 5.09770097,15.8903179 5.09770097,15.8903179 C6.10141502,15.9608925 6.62936247,16.9207077 6.62936247,16.9207077 C7.52134811,18.4481885 8.97013414,18.0069436 9.53982627,17.7510338 C9.63068234,17.1054292 9.88913104,16.6647979 10.174591,16.415025 C7.95414224,16.1627974 5.6195095,15.3048551 5.6195095,11.4741861 C5.6195095,10.3830406 6.00933116,9.49011794 6.64900703,8.79173579 C6.54587311,8.53889444 6.20270727,7.52200584 6.74722981,6.14610707 C6.74722981,6.14610707 7.5864207,5.87730972 9.4968538,7.17097368 C10.2943,6.94881697 11.150066,6.8383523 12.0003069,6.83405645 C12.849934,6.8383523 13.7050861,6.94881697 14.5037601,7.17097368 C16.4129654,5.87730972 17.2509285,6.14610707 17.2509285,6.14610707 C17.7966788,7.52200584 17.453513,8.53889444 17.350993,8.79173579 C17.9918966,9.49011794 18.3786488,10.3830406 18.3786488,11.4741861 C18.3786488,15.3146742 16.0403327,16.1597289 13.8131312,16.407047 C14.1716443,16.7157344 14.4914822,17.3257448 14.4914822,18.2585575 C14.4914822,19.5945663 14.4792044,20.6728241 14.4792044,21.000536 C14.4792044,21.2681059 14.6596888,21.5792481 15.1667639,21.481671 C19.1374198,20.1567087 22,16.4125702 22,11.9970521 C22,6.47565999 17.5222689,2 11.9990792,2"
                    fill-rule="nonzero"
                  />
                </svg>
              </a>
            </div>
            <div
              class="css-fcx7ij"
            >
              <a
                aria-label="twitter"
                class="css-5p18y4"
                data-testid="buttonLink"
                href="https://twitter.com/jpspenc"
              >
                <svg
                  aria-hidden="true"
                  class="css-1w2y819"
                  fill="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8.28984,21 C15.83736,21 19.96472,14.4602319 19.96472,8.78946539 C19.96472,8.60363327 19.96112,8.41863784 19.95296,8.23464647 C20.75408,7.62895589 21.45056,6.87316048 22,6.01260963 C21.26472,6.35415252 20.47352,6.58441367 19.64352,6.68816506 C20.49072,6.15702487 21.14112,5.31630393 21.44784,4.31426631 C20.65488,4.80583036 19.77704,5.1633543 18.8424,5.35588007 C18.09368,4.52176909 17.02728,4 15.84728,4 C13.58104,4 11.74336,5.92199451 11.74336,8.29129138 C11.74336,8.62814872 11.77944,8.95555129 11.84976,9.2698176 C8.4396,9.09051177 5.4156,7.38271368 3.392,4.78600045 C3.03928,5.42005532 2.83632,6.15702487 2.83632,6.94294166 C2.83632,8.43194145 3.56072,9.74632096 4.66224,10.5152526 C3.98928,10.493582 3.35688,10.3001358 2.80384,9.97833919 C2.80288,9.99624467 2.80288,10.0142338 2.80288,10.0330597 C2.80288,12.111769 4.21752,13.8469273 6.09488,14.2404296 C5.75032,14.3384914 5.3876,14.3913711 5.0132,14.3913711 C4.74888,14.3913711 4.49176,14.3640109 4.24184,14.3139759 C4.76424,16.0190129 6.27896,17.2597624 8.07512,17.2947366 C6.67048,18.4457914 4.90136,19.1317219 2.97888,19.1317219 C2.64776,19.1317219 2.3212,19.1119757 2,19.0723159 C3.81608,20.2894703 5.97224,21 8.28984,21"
                    fill-rule="nonzero"
                  />
                </svg>
              </a>
            </div>
            <div
              class="css-fcx7ij"
            >
              <button
                aria-label="enable night mode"
                class="css-5p18y4"
                data-testid="button"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="css-rbgw-EmotionIconBase ex0cdmw0"
                  fill="currentColor"
                  focusable="false"
                  overrides="[object Object]"
                  viewBox="0 0 48 48"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M37.9 33a18 18 0 0 1-30.4 1.23 1.35 1.35 0 0 1 .65-2.04c6.78-2.42 10.41-5.24 12.52-9.26 2.22-4.23 2.79-8.87 1.24-15.23a1.35 1.35 0 0 1 1.38-1.67A18 18 0 0 1 37.9 33.01z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    class="nk-layer-organizer"
  />
</body>
`;
